File: 0486-predict-the-winner/0486-predict-the-winner.java
class Solution {
    private boolean backtracking(int take, int left, int right,int playerOneScore,int playerTwoScore,int[] nums){
        if(left>right || right<left) return playerOneScore>=playerTwoScore;
        if(take%2==0){
            return backtracking(take+1,left+1,right,playerOneScore+nums[left],playerTwoScore,nums)||backtracking(take+1,left,right-1,playerOneScore+nums[right],playerTwoScore,nums);
        }
        return backtracking(take+1,left+1,right,playerOneScore,playerTwoScore+nums[left],nums) && backtracking(take+1,left,right-1,playerOneScore,playerTwoScore+nums[right],nums);
    }
    public boolean predictTheWinner(int[] nums) {
        return backtracking(0,0,nums.length-1,0,0,nums);
    }
}

File: 0486-predict-the-winner/0486-predict-the-winner.py
class Solution:
    def backtracking(self,take,left,right,playerOneScore,playerTwoScore,nums):
        if left>right or right<left:
            return playerOneScore>=playerTwoScore
        if take%2==0:
            return self.backtracking(take+1,left+1,right,playerOneScore+nums[left],playerTwoScore,nums) or self.backtracking(take+1,left,right-1,playerOneScore+nums[right],playerTwoScore,nums)
        return self.backtracking(take+1,left+1,right,playerOneScore,playerTwoScore+nums[left],nums) and self.backtracking(take+1,left,right-1,playerOneScore,playerTwoScore+nums[right],nums)
        
    def predictTheWinner(self, nums: List[int]) -> bool:
        return self.backtracking(0,0,len(nums)-1,0,0,nums)
        
        